lib/grant.js

```diff
@@ -460,6 +460,13 @@ function sendResponse (done) {
     access_token: this.accessToken
   };
 
+  if (this.model.addToResponse) {
+    this.model.addToResponse(this.req, function (err, extraInfo) {
```

Comment:
If we want this function to be asynchronous (ie., to take a callback rather than simply returning a value), then we'll need to change some more code down below to expect that.  As this currently stands, it will only work as expected if the callback is called _immediately_.  Otherwise, the `this.res.jsonp(response);` line down lower will get executed first.